src/containers/Profile/Profile.tsx

```diff
@@ -1,42 +1,47 @@
-import * as React from 'react'
+import React, { useCallback, useEffect, useMemo, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { Profile as BaseProfile } from 'decentraland-ui/dist/components/Profile/Profile'
+import { loadProfileRequest } from '../../modules/profile/actions'
+import { getData as getProfiles } from '../../modules/profile/selectors'
 import { Props } from './Profile.types'
 
-export default class Profile extends React.PureComponent<Props> {
-  static defaultProps = {
-    inline: true
-  }
+const Profile = function<T extends React.ElementType>(props: Props<T>) {
+  const { address, debounce, inline = true } = props
+  const profiles = useSelector(getProfiles)
+  const dispatch = useDispatch()
 
-  timeout: NodeJS.Timeout | null = null
+  const timeoutRef = useRef<NodeJS.Timeout>()
```

Comment:
Is it possible by any chance to use the type that the `setTimeout` function has in the dom or is it bothering at compile time?
```suggestion
  const timeoutRef = useRef<number>()
```